generators/client-2/templates/src/main/webapp/app/blocks/handlers/_state.handler.ts

```diff
@@ -35,25 +36,27 @@ export function StateHandler($rootScope, $state, $sessionStorage, <% if (enableT
                 $translate.use(language);
             });*/
             <% } %>
-        });
+        }));
 
-        var stateChangeSuccess = $rootScope.$on('$stateChangeSuccess',  function(event, toState, toParams, fromState, fromParams) {
+        // Redirect to a state with an external URL (http://stackoverflow.com/a/30221248/1098564)
+        deregistrationFns.push($transitions.onStart({ to: state => state.external }, (transition: Transition) => {
```

Comment:
note how the  `{ to: function(state) { return !!state.external } }` criteria object only runs the hook for matching transitions.